aioauth/errors.py

```diff
@@ -82,15 +82,15 @@ class InvalidClientError(OAuth2Error[TRequest]):
     client.
     """
 
-    error: Literal["invalid_client"] = "invalid_client"
+    error: ErrorType = "invalid_client"
```

Comment:
I wanted to ask you about this. What do you think about loosening the type of `OAuth2Error.error` from `ErrorType` to `str`? It would make it easier for clients of the library to reuse `OAuth2Error` for custom messages. 

I came across the issue while trying to implement some of the error cases that come up with the `prompt` argument to the authorization endpoint that comes with OIDC. For example, one of the error cases is when the user is not signed in. In that scenario, the server should respond [with an error, typically of `login_required`](https://openid.net/specs/openid-connect-core-1_0.html#AuthError:~:text=MUST%20return%20an%20error%2C%20typically%20login_required.).

Given how the library currently expects user authentication to be handled independently, it seemed like the code to raise an error in that scenario might not be code that belonged in this library. In such a case, it would be handy to be able repurpose the `OAuth2Error` class to raise a more consistent error.

That said, even with the OIDC extensions, perhaps there are few enough error codes that you'd rather keep the strict typing. There may be others out there, but OIDC recommends at least `account_selection_required`, `consent_required`, `login_required`, `interaction_required` and [a few others](https://openid.net/specs/openid-connect-core-1_0.html#AuthError)

Reply:
Very good point I thought about that too. But the downside is that there is little chance that users of this library will extend it, since this library implements already existing standards.

> There may be others out there, but OIDC recommends at least account_selection_required, consent_required, login_required, interaction_required and [a few others](https://openid.net/specs/openid-connect-core-1_0.html#AuthError)

In your PR I mentioned that probably it make sense to start working on OIDC support under the feature flag as one of the possible solutions.

Reply:
I'm not sure where I stand yet. For now, trying to live with some of the code I've written and see how it ages. That said, as far as I'm concerned, this point is not a reason to delay merging this PR.